src/components/Schedule/AppointmentModal.tsx

```diff
@@ -41,11 +41,33 @@ const USERS = gql`
   }
 `;
 
-interface ProfessorOption extends OptionBase {
+const GET_COURSES = gql`
+  query GetCourses {
+    survey {
+      courses {
+        subject
+        code
+        term
+        title
+      }
+    }
+  }
+`;
+
+interface Options extends OptionBase {
   label: string;
   value: string;
 }
 
+interface CourseOptions extends OptionBase {
+  label: string;
+  value: {
+    code: string;
+    subject: string;
+    title: string;
+  };
```

Comment:
I think you can do this
```suggestion
  value: CourseID;
```

Reply:
or is CourseID the one with `term` lol I can't remember

Reply:
yeah CourseID includes term, and we don't support editing of terms so I left it out